packages/vuetify/src/components/VApp/VApp.tsx

```diff
@@ -18,10 +19,12 @@ export const VApp = defineComponent({
   },
 
   setup (props, { slots }) {
-    const { themeClasses } = useTheme(props)
+    const { themeClasses, current } = useTheme()
     const { layoutClasses, getLayoutItem, items } = createLayout(props)
     const { rtlClasses } = useRtl()
 
+    watch(() => props.theme, value => value && (current.value = value), { immediate: true })
```

Comment:
VApp isn't a singleton anymore, you can have more than one which could cause some problems with this watcher. Any reason not to just use `provideTheme` instead?

Reply:
Isn't it still kind of? What's an example of where you would have multiple ones?

The reason for this change is so that you can call `useTheme` where you are rendering VApp and it will be the same theme context. Perhaps a pretty niche thing so I wouldn't be too sad to revert it 🤷

Reply:
Yeah seems pretty niche. You could also expose `this.$refs.app.theme` for the same effect.